src/pageEditor/modListingPanel/DraftModComponentListItem.tsx

```diff
@@ -121,15 +119,17 @@ const DraftModComponentListItem: React.FunctionComponent<
       showConfirmationModal: DEACTIVATE_MOD_MODAL_PROPS,
     });
 
-  const onSave = async () => {
-    if (modComponentFormState.modMetadata) {
-      await saveMod(modComponentFormState.modMetadata?.id);
-    } else {
-      dispatch(actions.showCreateModModal({ keepLocalCopy: false }));
+  const onSave = useMemo(() => {
+    if (modComponentFormState.modMetadata == null) {
+      return async () => {
+        dispatch(actions.setActiveModComponentId(modComponentFormState.uuid));
+        dispatch(actions.showCreateModModal({ keepLocalCopy: false }));
+      };
     }
-  };
 
-  const isSaving = modComponentFormState.modMetadata ? isSavingMod : false;
+    // eslint-disable-next-line unicorn/no-useless-undefined -- Code clarity, implicit returns are bad
+    return undefined;
```

Comment:
Why not simply return?
```suggestion
    return;
```

Reply:
This also requires suppressing a lint warning. Do you like this better in terms of code clarify or was this suggestion an effort to avoid the lint check?

Reply:
So we have conflicting lint rules? We should address that in a separate PR

> was this suggestion an effort to avoid suppressing the lint check?

☝️